File: LeetCode/python/leet_0334.py
from typing import List

class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        ma = max(nums)
        mi = min(nums)
        mil = []
        mal = []
        count = 0
        for num in nums:
            mal.append(ma)
            ma = min(ma, num)
        for num in nums[::-1]:
            mil.append(mi)
            mi = max(mi, num)
        mil = mil[::-1]
        print(mal)
        print(mil)
        for i in range(len(nums)):
            if nums[i] > mal[i] and nums[i] < mil[i]:
                return True
        return False

        

File: LeetCode/leet_0334.cpp
class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {
        vector<int> prefix;
        vector<int> sufix(nums.size());
        int mi = INT_MAX;
        int ma = 0;
        for(int i =0; i<nums.size(); i++){
            mi = min(nums[i], mi);
            prefix.push_back(mi);
        }
        for(int i =nums.size()-1; i>=0; i--){
            ma = max(nums[i], ma);
            sufix[i] = ma;
        }
        for(int i = 0; i<nums.size(); i++){
            if(prefix[i]<nums[i] && nums[i]<sufix[i])
                return true;
        }

        return false;
    }
};